### DISCUSSION TITLE
audit trail

### DISCUSSION THREAD

**Discussion by nyemiks** [opencrx/opencrx#96] [Q&A] [answered]
Hi Everyone,

From the opencrx site it says it has a pervasive audit trail system. I will like to ask the following questions:

1. Which uml model handles audit trails in opencrx ?

2.  How can I view the audit trail for a given user ? Would I appreciate sample code snippets ? (java and/or sql)

3. Which tables store the audit trail information ?

Regards,

**Comment by wfro64** [comment, marked as answer]
Here are some answers:
* Model: see [here](https://www.opencrx.org/opencrx/5.1/uml/opencrx-core/base/tn/999_Audit.png.html). Moreover, every CrxObject inherits from Auditee as shown [here](https://www.opencrx.org/opencrx/5.1/uml/opencrx-core/generic/tn/020_CrxObject.png.html), i.e. every update on a CrxObject is tracked as AuditEntry
* In the standard GUI you can see the audit trail for a given object in the tab / grid 'Audit Entries'. There is no tab / grid which shows all AuditEntries for a given user
* The audit trail is stored in table OOCKE1_AUDITENTRY. You can select entries for a given user and / or changed object: `select * from oocke1_auditentry ae inner join oocke1_auditentry_ ae_ on ae.object_id = ae_.object_id where ae_.created_by = 'guest'`

**Reply by nyemiks** [reply]
Thanks for this explanation.
